### PR TITLE
Extend the readme with code examples

### PR DESCRIPTION
**Pull request by msm-code**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,29 +1,111 @@
...
-============
+## Features
 
-* Support for (non)memory-mapped PE images without header fix-up.
-* Searching for wildcarded byte sequences
-* Support for x64 disassembly
-* Fixed-precision integer types
-* Many improvements in ProcessMemory
+- **Cryptography** (AES, Blowfish, Camelie, ChaCha20, Serpent and many others)
+- **Compression algorithms** (aplib, gzip, inznt1 (RtlDecompressBuffer)
```
Comment (nazywam) [inline README.md:11]:
```suggestion
- **Compression algorithms** (aplib, gzip, inznt1 (RtlDecompressBuffer))
```

File: README.md [hunk 2]
```diff
@@ -1,29 +1,111 @@
...
+
+### Serpent
+
+```python
+from malduck import serpent
+
+key = b'a'*16
+iv = b'b'*16
+plaintext = b'data'*16
+ciphertext = serpent.cbc.encrypt(key, plaintext, iv=iv)
```
Comment (nazywam) [inline README.md:39]: It seems kinda weird to use kwargs here and nowhere else.

File: README.md [hunk 3]
```diff
@@ -1,29 +1,111 @@
...
+# Headerless compressed buffer
+aplib(b'T\x00he quick\xecb\x0erown\xcef\xaex\x80jumps\xed\xe4veur`t?lazy\xead\xfeg\xc0\x00')
+```
+
+### Fixed integer types
+
+```python
+from malduck import DWORD
+
+def sdbm_hash(name: bytes):
```
Comment (nazywam) [inline README.md:56]:
```suggestion
def sdbm_hash(name: bytes) -> int:
```

File: README.md [hunk 4]
```diff
@@ -1,29 +1,111 @@
...
+            return {'login_key': p.asciiz(hit)}
+```
+
+### Memory model objects
+
+```python
+from malduck import procmempe
+
+with procmempe.from_file("notepad.exe", image=True) as p:
+    p.pe.resource("NPENCODINGDIALOG")
```
Comment (nazywam) [inline README.md:98]:
```suggestion
    resource_ data = p.pe.resource("NPENCODINGDIALOG")
```

**Review by nazywam — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -1,29 +1,111 @@
...
+### Fixed integer types
+
+```python
+from malduck import DWORD
+
+def sdbm_hash(name: bytes):
+    hh = 0
+    for c in name:
+        hh = DWORD(c) + (hh << 6) + (hh << 16) - hh
+    return hh
```
Comment (nazywam) [inline README.md:60]:
```suggestion
    return int(hh)
```

File: README.md [hunk 6]
```diff
@@ -1,29 +1,111 @@
...
+
+### Fixed integer types
+
+```python
+from malduck import DWORD
+
+def sdbm_hash(name: bytes):
+    hh = 0
+    for c in name:
+        hh = DWORD(c) + (hh << 6) + (hh << 16) - hh
```
Comment (nazywam) [inline README.md:59]: I think a small comment here that explains how the DWORD type is propagated to the rest of the variables could be useful

File: README.md [hunk 7]
```diff
@@ -1,29 +1,111 @@
...
+## How to start
+
+Install it by running
 
 ```
 pip install malduck
 ```
 
-Usage documentation can be found [on readthedocs](https://malduck.readthedocs.io/en/latest/).
+More documentation can be found [on readthedocs](https://malduck.readthedocs.io/en/latest/).
```
Comment (nazywam) [inline README.md:109]:
```suggestion
More documentation can be found on [readthedocs](https://malduck.readthedocs.io/en/latest/).
```

**Review by nazywam — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (msm-code) [reply README.md]: I've actually mostly copied the examples from the docs. Do you think that it's a good idea to back-port your suggestions to the code examples in the main docs?

Comment (msm-code) [reply README.md]: Other weird thing: cbc.encrypt is (key, iv, data), and serpent.encrypt is (key, data, iv) :thinking:

File: README.md (hunk 3 above)
Comment (msm-code) [reply README.md]: wait, it's actually a DWORD, not an int. Fixed.

File: README.md [hunk 8]
```diff
@@ -1,29 +1,113 @@
...
+## Features
 
-* Support for (non)memory-mapped PE images without header fix-up.
-* Searching for wildcarded byte sequences
-* Support for x64 disassembly
-* Fixed-precision integer types
-* Many improvements in ProcessMemory
+- **Cryptography** (AES, Blowfish, Camelie, ChaCha20, Serpent and many others)
+- **Compression algorithms** (aPLib, gzip, LZNT1 (RtlDecompressBuffer))
+- **Memory model objects** (work on memory dumps, PE/ELF, raw files and IDA dumps with the same code)
```
Comment (nazywam) [inline README.md:12]:
```suggestion
- **Memory model objects** (work on memory dumps, PE/ELF, raw files and IDA dumps using the same code)
```

File: README.md [hunk 9]
```diff
@@ -1,29 +1,113 @@
...
+### Fixed integer types
+
+```python
+from malduck import DWORD
+
+def sdbm_hash(name: bytes) -> int:
+    hh = 0
+    for c in name:
+        # operations on the DWORD type produce a dword, so a result
+        # is also a DWORD.
```
Comment (nazywam) [inline README.md:60]:
```suggestion
        # hh is also a DWORD.
```

**Review by nazywam — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (nazywam) [reply README.md]: Hmh, doesn't seem good. I don't have any ideas how to fix it without breaking compatiblity though.

File: README.md (hunk 9 above)
Comment (msm-code) [reply README.md:60]: doesn't sound good (`so a result hh is also a DWORD`). `so as a result`?

**Pull request merged**